### DISCUSSION TITLE
ESP32_TOUCH

### DISCUSSION THREAD

**Discussion by nuhaildam** [Mair/esp32-course#86] [Course Qeustions] [answered]
E
![Touch_code_error](https://user-images.githubusercontent.com/104188092/164613856-53cabc83-9334-4345-ada4-22a3ef1bd354.png)
sp32 touch example in the course is not working.
It gives the following error:
../main/main.c: In function 'app_main':
../main/main.c:10:22: error: 'TOUCH_PAD_GPIO13_CHANNEL' undeclared (first use in this function); did you mean 'TOUCH_PAD_SLOPE_MAX'?
     touch_pad_config(TOUCH_PAD_GPIO13_CHANNEL, -1);
                      ^~~~~~~~~~~~~~~~~~~~~~~~
                      TOUCH_PAD_SLOPE_MAX
../main/main.c:10:22: note: each undeclared identifier is reported only once for each function it appears in
ninja: build stopped: subcommand failed.
ninja failed with exit code 1

**Comment by Mair** [comment, marked as answer]
@nuhaildam 
Thanks for pointing this out.
Yes there was an update on the IDF that has changed this.
I have updated the video. please see  https://learnesp32.com/videos/9/10_touch at 3:52

In short to correct this I added this line 
```c
#define TOUCH_PAD_GPIO13_CHANNEL TOUCH_PAD_NUM4
```

full source 

```C
#include <stdio.h>
#include "freertos/FreeRTOS.h"
#include "freertos/task.h"
#include "driver/touch_pad.h"

#define TOUCH_PAD_GPIO13_CHANNEL TOUCH_PAD_NUM4

void app_main()
{
    touch_pad_init();
    touch_pad_set_voltage(TOUCH_HVOLT_2V7, TOUCH_LVOLT_0V5, TOUCH_HVOLT_ATTEN_1V);
    touch_pad_config(TOUCH_PAD_GPIO13_CHANNEL, -1);
    touch_pad_filter_start(10);

    uint16_t val;
    uint16_t touch_filter_value = 0;
    uint16_t touch_rawValue = 0;

    while (true)
    {
        touch_pad_read_raw_data(TOUCH_PAD_GPIO13_CHANNEL, &touch_rawValue);
        touch_pad_read_filtered(TOUCH_PAD_GPIO13_CHANNEL, &touch_filter_value);
        touch_pad_read(TOUCH_PAD_GPIO13_CHANNEL, &val);
        printf("val = %d raw = %d filtered = %d\n", val, touch_rawValue, touch_filter_value);
        vTaskDelay(1000 / portTICK_PERIOD_MS);
    }
}
```
